clustermesh-apiserver/main.go

```diff
@@ -18,13 +19,21 @@ import (
 	"github.com/cilium/cilium/pkg/hive"
 )
 
-func main() {
-	cmd := &cobra.Command{
-		Use:   "clustermesh-apiserver",
-		Short: "Run the ClusterMesh apiserver",
-	}
+var cmdRefDir string
+
+var rootCmd = &cobra.Command{
+	Use:   "clustermesh-apiserver",
+	Short: "Run the ClusterMesh apiserver",
+}
 
-	cmd.AddCommand(
+func init() {
+	flags := rootCmd.PersistentFlags()
+	flags.StringVar(&cmdRefDir, "cmdref", "", "Path to cmdref output directory")
+	flags.MarkHidden("cmdref")
+}
```

Comment:
Is there any reason for controlling this functionality via a flag, rather than a dedicated `cmdref` command? That looks more idiomatic and less error-prone to me. 

Ideally, all binaries could also reuse a common `cmdref` command ([here](https://github.com/cilium/cilium/blob/main/daemon/cmd/root.go#L51-L59) is the agent one, for instance), to ensure consistency and reduce code duplication, but that can definitely be a follow-up.

Reply:
I mapped this implementation on the one from [cilium-health](https://github.com/cilium/cilium/blob/ca18209dd67a5b6e8e3cf6b114441cc9495d37e3/cilium-health/cmd/root.go#L64), which uses a flag, I can update it to use a subcommand instead.

Reply:
Updated in [0dad7b5](https://github.com/cilium/cilium/pull/36205/commits/0dad7b5b018ef1857124b27eaf9dbe9350a8236c)